conans/test/integration/toolchains/cmake/cmakedeps/test_cmakedeps.py

```diff
@@ -707,3 +707,35 @@ def generate(self):
     assert 'set(dep_NO_SONAME_MODE_RELEASE TRUE)' in dep
     other = c.load("app/other-release-data.cmake")
     assert 'set(other_other_mycomp1_NO_SONAME_MODE_RELEASE TRUE)' in other
+
+def test_cmakedeps_set_legacy_variable_name():
+    client = TestClient()
+    conanfile = str(GenConanfile("dep", "1.0"))
+    conanfile += """
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "CMakeFileName")
+    """
+    client.save({"dep/conanfile.py": conanfile})
+    client.run("create dep")
+    client.save({"conanfile.py": GenConanfile("lib", "1.0")
+                .with_requires("dep/1.0").with_generator("CMakeDeps").with_setting("build_type")})
+    client.run("install .")
+    dep_config = client.load("CMakeFileNameConfig.cmake")
+    cmake_variables = ["VERSION_STRING", "INCLUDE_DIRS", "INCLUDE_DIR", "LIBRARIES", "DEFINITIONS"]
+    for variable in cmake_variables:
+        assert f"CMakeFileName_{variable}" in dep_config
+
+    conanfile = str(GenConanfile("dep", "1.0"))
+    conanfile += """
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "NewCMakeFileName")
+        self.cpp_info.set_property("cmake_additional_variables_prefixes", ["PREFIX", "prefix", "PREFIX"])
+    """
+    client.save({"dep/conanfile.py": conanfile})
+    client.run("create dep")
+    client.run("install .")
```

Comment:
If you do the simplification suggested above you can do the same here:

```suggestion
    client.run("install --requires=dep/1.0 -g CMakeDeps")
```